### PR TITLE
Add 256 byte block hamming code implementation

### PR DESCRIPTION
**Pull request by x3ro**
I'll need this for the lower layer of the file system thingy I'm building :smile: Found the code [here](https://code.google.com/p/android-source-browsing/source/browse/MakefileBasedBuild/Atmel/sam3x/sam3x-ek/libraries/libboard_sam3x-ek/source/hamming.c?repo=device--google--accessory--adk2012&r=e0f114ab1d645caeac2c30273d0b693d72063f54), seems license compatible with what we have, right?

### CODE REVIEW

**Comment by OlegHahm** [comment]
License seems compatible to me, yes.

**Comment by OlegHahm** [comment]
Can you update the code to match our coding conventions, maybe using [uncrustify](https://github.com/RIOT-OS/RIOT/pull/4063)?

**Comment by x3ro** [comment]
Done :)

File: sys/ecc/hamming256/hamming256.c [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+ */
+
+
+/*----------------------------------------------------------------------------
+ *        Headers
+ *----------------------------------------------------------------------------*/
+
+#include <stdint.h>
+
+#define ENABLE_DEBUG (1)
```
Comment (OlegHahm) [inline sys/ecc/hamming256/hamming256.c]: should be off by default

File: sys/ecc/hamming256/hamming256.c [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+
+/*----------------------------------------------------------------------------
+ *         Internal function
+ *----------------------------------------------------------------------------*/
+
+/**
+ *  Counts and return the number of bits set to '1' in the given byte.
+ *  \param byte  Byte to count.
+ */
+static uint8_t CountBitsInByte(uint8_t byte)
```
Comment (OlegHahm) [inline sys/ecc/hamming256/hamming256.c]: Use `bitarithm_bits_set` instead.

**Comment by OlegHahm** [comment]
Please use also C style comments and update the code to match RIOT naming conventions (minuscules instead of capitals) and use the `@` for doxygen tags.

File: sys/ecc/hamming256/hamming256.c [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+{
+    uint32_t i;
+    uint8_t columnSum = 0;
+    uint8_t evenLineCode = 0;
+    uint8_t oddLineCode = 0;
+    uint8_t evenColumnCode = 0;
+    uint8_t oddColumnCode = 0;
+
+    // Xor all bytes together to get the column sum;
+    // At the same time, calculate the even and odd line codes
```
Comment (OlegHahm) [inline sys/ecc/hamming256/hamming256.c]: Can you change these comments to C style (`/* .. */`), too?

**Comment by x3ro** [comment]
Like this?

File: sys/include/ecc/hamming256.h [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+#define Hamming_ERROR_ECC               2
+
+/** Multiple bits are incorrect in the data and they cannot be corrected. */
+#define Hamming_ERROR_MULTIPLEBITS      3
+
+
+/**
+ *  Computes 3-bytes hamming codes for a data block whose size is multiple of
+ *  256 bytes. Each 256 bytes block gets its own code.
+ *  \param data  Data to compute code for.
```
Comment (OlegHahm) [inline sys/include/ecc/hamming256.h]: Can you add `@brief` and change `\param` to `@param[in]` (or `out`), please?

Comment (x3ro) [inline sys/include/ecc/hamming256.h]: Why `@brief`? Isn't the first line until the full-stop automatically the brief desc?

Comment (OlegHahm) [inline sys/include/ecc/hamming256.h]: At least for consistency.

**Comment by x3ro** [comment]
Build is passing. You think this is ready to be merged after squashing?

**Comment by OlegHahm** [comment]
Sorry for being nitpicky, but have you ran uncrustify or something like this? There are some superfluous whitespaces after opening and before closing parens.

**Comment by x3ro** [comment]
Yep, did run uncrustify with the RIOT settings things.

**Comment by OlegHahm** [comment]
Okay, there were some options missing. Try x3ro/RIOT#4

**Comment by x3ro** [comment]
Done

**Comment by OlegHahm** [comment]
ACK

**Pull request merged**